projects/batfish-common-protocol/src/main/java/org/batfish/storage/FileStorageImpl.java

```diff
@@ -0,0 +1,158 @@
+    Path aDir =
+        _containersLocation.resolve(
+            Paths.get(containerName, BfConsts.RELPATH_ANALYSES_DIR, analysisName));
+    if (!Files.exists(aDir)) {
+      throw new BatfishException(
+          String.format(
+              "Analysis '%s' doesn't exist for container '%s'", analysisName, containerName));
+    }
+    Path questionsDir = aDir.resolve(BfConsts.RELPATH_QUESTIONS_DIR);
+    if (!Files.exists(questionsDir)) {
+      throw new BatfishException(
+          String.format("Analysis '%s' doesn't contain questions directory", analysisName));
+    }
+    Map<String, String> questions = new HashMap<>();
+    for (Path questionDir : CommonUtil.getEntries(questionsDir)) {
+      Path questionFile = questionDir.resolve(BfConsts.RELPATH_QUESTION_FILE);
+      if (!Files.exists(questionFile)) {
+        throw new BatfishException(
+            String.format(
+                "Failed to read analysis '%s', question empty: '%s'",
+                analysisName, questionDir.getFileName()));
+      }
+      questions.put(
+          questionDir.getFileName().toString(),
+          CommonUtil.readFile(questionFile));
+    }
+    return new Analysis(analysisName, questions);
+  }
+
+  @Override
+  public Analysis saveAnalysis(String containerName, Analysis analysis) {
+    Path aDir =
+        _containersLocation.resolve(
+            Paths.get(containerName, BfConsts.RELPATH_ANALYSES_DIR, analysis.getName()));
+    if (Files.exists(aDir)) {
+      throw new BatfishException(
+          String.format(
+              "Analysis '%s' already exists for container '%s'",
+              analysis.getName(), containerName));
+    }
+    //trying to create analysis skeleton dir with questions dir
+    if (!aDir.resolve(BfConsts.RELPATH_QUESTIONS_DIR).toFile().mkdirs()) {
+      throw new BatfishException(String.format("Failed to create analysis directory '%s'", aDir));
+    }
+
+    for (String questionName : analysis.getQuestions().keySet()) {
+      Path questionDir = aDir.resolve(Paths.get(BfConsts.RELPATH_QUESTIONS_DIR, questionName));
+      if (!questionDir.toFile().mkdirs()) {
+        throw new BatfishException(
+            String.format("Failed to create question directory '%s'", questionDir));
+      }
+      Path questionFile = questionDir.resolve(BfConsts.RELPATH_QUESTION_FILE);
+      CommonUtil.writeFile(questionFile, analysis.getQuestions().get(questionName));
+    }
+
+    return getAnalysis(containerName, analysis.getName());
+  }
+
+  @Override
+  public Analysis updateAnalysis(String containerName, Analysis analysis) {
+    Analysis oldAnalysisObj = getAnalysis(containerName, analysis.getName());
+    Set<String> questionsToDelete =
+        Sets.difference(oldAnalysisObj.getQuestions().keySet(), analysis.getQuestions().keySet());
+    Set<String> questionsToAdd =
+        Sets.difference(analysis.getQuestions().keySet(), oldAnalysisObj.getQuestions().keySet());
+    Path questionsDir =
```

Comment:
I think it would be nice to have all private function calls to get the directories. When the number of function grows in this class, using Path.resolve everywhere will make it hard to manage.

Reply:
Done, separated the functions to get various paths in a StorageUtils class